ships/0026-image-abstraction-layer.md

```diff
@@ -0,0 +1,232 @@
+| address    | The mirror registry URL                                                           |
+| username   | Username                                                                          |
+| password   | The password to be used                                                           |
+| token      | The auth token to be used                                                         |
+| insecure   | Allows access to insecure registries if set to "true"                             |
+| repository | If set all images are going to be mirrored inside this registry repository        |
+
+### Image status
+
+The status field for a given Image should contain all its previously imported versions of the
+same image sorted by import time. Follow below the proposed properties found in an Image `.status`
+property and their meaning:
+
+| Name              | Description                                                                |
+| ----------------- | -------------------------------------------------------------------------- |
+| hashReferences    | A list of all imported versions of the Image. Sorted by import time descending (the first item in this list is the last imported version) |
+   
+The property `.status.hashReferences` is a list of imported image versions, The operator would
+hold up to twenty five imports per image. Each item in the list is composed of the following
+properties:
+
+| Name           | Description  |
+| -------------- | ------------ |
+| from           | Keeps a reference from where the image got imported |
+| importedAt     | Date and time of the import |
+| imageReference | Where this reference points to (by hash), may point to the mirror registry in case of a mirrored Image |
+
+### ImageImport status
+
+On a ImageImport status, one would be able to find information about the import attempts executed
+by the operator, it looks like:
+
+| Name           | Description  |
+| -------------- | ------------ |
+| hashReference  | Once the import has succeeded this is populated with the data about the image. Properties for hashReference are equal to Image.Status.HashReferences above. This is what is copied into the Image.Status.HashReferences list. |
+| importedAt     | A list of failures or successes during the ImageImport processing |
+
+The properties for the object `.status.hashReference` are exactly the same as for the objects
+inside `Image.Status.HashReferences`. As for importAttempts objects they contain the following
+properties:
+
+| Name    | Description                                                                          |
+| ------- | ------------------------------------------------------------------------------------ |
+| when    | Date and time of the import attempt                                                  |
+| succeed | A boolean indicating if the import was successful or not                             |
```

Comment:
Would this be rather a status (True, False, Unknown)?

Reply:
Yes ! You are absolutely right ! This should be a Condition inside the Status. I am going to address it.

Reply:
I have adjusted this EP to include a "Condition" property in the ImageImport status. PTAL.